milvus/provider/client.py

```diff
@@ -0,0 +1,107 @@
+import logging
+from flask import current_app as app
+import cohere
+from pymilvus import connections, Collection
+
+logger = logging.getLogger(__name__)
+
+cohere_client = None
+
+
+class MilvusConnectorClient:
+    DEFAULT_SEARCH_LIMIT = 10
+    DEFAULT_ALIAS = "default"
+
+    def __init__(
+        self,
+        cluster_uri,
+        collection_name,
+        vector_field,
+        user=None,
+        password=None,
+        api_key=None,
+    ):
+        if api_key is not None:
+            connections.connect(
+                alias=self.DEFAULT_ALIAS, uri=cluster_uri, token=api_key
+            )
+        else:
+            connections.connect(
+                alias=self.DEFAULT_ALIAS, uri=cluster_uri, token=f"{user}:{password}"
+            )
+        self.collection = Collection(name=collection_name)
+        self.vector_field = vector_field
+        self.search_limit = self.DEFAULT_SEARCH_LIMIT
+
+    def set_search_limit(self, search_limit):
```

Comment:
I would remove the setter, either you can pass the search_limit into the `__init__` method or just set the search_limit attribute directly, in general setting/getting in Python is not recommended unless you have specific reasons to do so

Reply:
Thank you. Done